r/R/dplyr.R

```diff
@@ -520,6 +536,60 @@ arrow_stringr_string_replace_function <- function(FUN, max_replacements) {
   }
 }
 
+# this function assigns definitions for the stringr pattern modifier functions
+# (fixed, regex, etc.) plus a required helper function (ensure_opts) in the
+# calling function's environment, where they are required to evaluate `pattern`
```

Comment:
This is a bit exotic, I know. I can't think of any cleaner or more concise way to assign these helper functions in both of the functions that need them.

Reply:
Why not pull `ensure_opts` out, call it `ensure_stringr_opts <- function(pattern)`, and have your funky eval inside that?

Reply:
D'oh, that seem so obvious now 🤦‍♂️ Thanks

Reply:
Done in 5bf0cf4148d2c572b00ca1e998d5a0590d4e5e5d